src/modvault/moditem.py

```diff
@@ -0,0 +1,211 @@
+import os
+import util
+from PyQt5 import QtCore, QtWidgets, QtGui
+from downloadManager import IconCallback
+import urllib.parse
+
+
+class ModView(QtCore.QObject):
+    """
+    Helps with displaying mods in the mod widget. Handles updates to view
+    unrelated to underlying data, like downloading mod previews. Forwards
+    interaction with the view.
+    """
+    mod_double_clicked = QtCore.pyqtSignal(object)
+
+    def __init__(self, model, view, delegate, dler):
+        QtCore.QObject.__init__(self)
+        self._model = model
+        self._view = view
+        self._delegate = delegate
+        self._dler = dler
+
+        self._view.setModel(self._model)
+        self._view.setItemDelegate(self._delegate)
+        self._delegate.mod_preview_missing.connect(self.download_mod_preview)
+        self._view.doubleClicked.connect(self._mod_double_clicked)
+        self._view.viewport().installEventFilter(self._delegate.tooltip_filter)
+
+    def download_mod_preview(self, mod):
+        name = os.path.basename(urllib.parse.unquote(mod.thumbnail))
+        cb = IconCallback(name, self._mod_preview_downloaded)
+        self._dler.downloadModPreview(mod.thumbnail, name, cb)  # (url, name, requester)
+
+    # TODO make it a utility function?
+    def _model_items(self):
```

Comment:
IIRC we don't actually use this, so it can go. Something to be fixed in GameView :/

Reply:
So remove the whole Preview part, will do.

Reply:
Wait, I meant just the function. Maybe I confused it with some other one, sorry :/ If it's used anywhere, leave it.

Reply:
As I mentioned in the description (remark 1), the whole preview downloader part is currently unused, because the icon is set in ModModelItem, but without downloading missing icons.
(As you mentioned the preview in gameitem was probably a bad idea, I left this up for debate.)

Reply:
Alright. I'll just make needed fixes in my downloader refactor branch after we merge this.

Reply:
You fix the downloader refactor, (and after that, I can adapt the downloader into the modvault refactor) and we merge this
or:
We merge this and after that you'll make the needed downloader fixes.
which is it?

Reply:
Your changes are more localized, so I guess I'll merge my fixes first. Either way we'll have to wait just a little bit (a week or so) for 0.16.1 to stabilize for good.